src/Microsoft.Restier.Core/Query/QueryModelReference.cs

```diff
@@ -154,18 +177,26 @@ public IEdmElement Element
     }
 
     /// <summary>
-    /// Represents a reference to derived data in terms of a model.
+    /// Represents a reference to property data in terms of a model.
     /// </summary>
-    public class DerivedDataReference : QueryModelReference
+    public class PropertyModelReference : QueryModelReference
     {
+        private readonly string propertyName;
+
         /// <summary>
-        /// Initializes a new instance of the <see cref="DerivedDataReference" /> class.
+        /// Initializes a new instance of the <see cref="PropertyModelReference" /> class.
         /// </summary>
         /// <param name="source">
         /// A source query model reference.
         /// </param>
-        public DerivedDataReference(QueryModelReference source)
+        /// <param name="propertyName">
+        /// The name of a property.
+        /// </param>
+        public PropertyModelReference(QueryModelReference source, string propertyName)
```

Comment:
Mark the ctor internal: 1) keep consistent; 2) we don't want user to be able to instantiate tis class.